From d69aaf2e7fb97aca301428e6533de934b970994f Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Fri, 19 Mar 2021 05:58:28 +0200
Subject: [PATCH] [SW-34719] gaudi: rma: set boot error bit

Incase of power cycle or warm reset the boot error
SP register gets erased. To recover the error status,
if fatal error found then boot error bit is set explicitly.

Change-Id: I2adc5a08a85cddfb36468b6c4f842b591527ea2c
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/gaudi/rma_info.c b/gaudi/rma_info.c
index 8d0a233..59c580e 100644
--- a/gaudi/rma_info.c
+++ b/gaudi/rma_info.c
@@ -11,6 +11,7 @@
 #include <string.h>
 #include "uart.h"
 #include "rma_if.h"
+#include "boot_if.h"
 #include "module_i2c_slave.h"
 #include "hl_i2c_slave_events.h"
 
@@ -174,6 +175,8 @@
 		rma_header->rma_failure.row_replacement ||
 		rma_header->rma_failure.hbm_init) {
 		err_sts_reg_val |= HL_ERROR_STATUS_FATAL_ERROR;
+		/* set boot error bit for RMA */
+		cpu_boot_err_clrset(CPU_BOOT_ERR0_DEVICE_UNUSABLE_FAIL, true);
 		preboot_puts_hex("Fatal error:", rma_header->rma_failure.data, 0);
 	}
 }
